src/platforms/esp32/components/avm_sys/platform_nifs.c

```diff
@@ -77,6 +81,18 @@ static const char *const esp_rst_brownout       = "\x10" "esp_rst_brownout";
 static const char *const esp_rst_sdio           = "\xC"  "esp_rst_sdio";
 //                                                        123456789ABCDEF01
 
+enum NetworkInterface {
+    WifiSTAInterface,
+    WifiSoftAPInterface,
+    InvalidInterface
+};
+
+static const AtomStringIntPair interface_table[] = {
+    { ATOM_STR("\x8", "wifi_sta"), WifiSTAInterface },
+    { ATOM_STR("\xB", "wifi_softap"), WifiSoftAPInterface },
+    SELECT_INT_DEFAULT(InvalidInterface)
+};
```

Comment:
Did you try doing:
`{ ATOM_STR("\x8", "wifi_sta"), ESP_MAC_WIFI_STA },`.
I think it might avoid you doing a switch and you can just check if != dummy value (likely -1 might work).

Reply:
That seems incredibly dangerous.  How do you know that ESP_MAC_EIFI_STA will always be an enumerated (or integral) type?  That is just setting yourself up for failure in the future.

Reply:
This is a fair point indeed.